Lib/test/test_argparse.py

```diff
@@ -4958,6 +4958,51 @@ def custom_type(string):
     version = ''
 
 
+class TestHelpUsageLongSubparserCommand(TestCase):
+    """Test that subparser commands are formatted correctly in help"""
+
+    def test_parent_help(self):
+        def custom_formatter(prog):
+            return argparse.RawTextHelpFormatter(prog, max_help_position=50)
+
+        parent_parser = argparse.ArgumentParser(
+                prog='PROG',
+                add_help=False,
+                formatter_class=custom_formatter
+        )
+        main_parser = argparse.ArgumentParser(
+            formatter_class=custom_formatter)
+
+        cmd_subparsers = main_parser.add_subparsers(title="commands",
+                                                    metavar='CMD',
+                                                    help='command to use')
+        cmd_parser = cmd_subparsers.add_parser("add",
+                                               help="add something",
+                                               parents=[parent_parser])
+        cmd_parser.add_subparsers(title="action", dest="action_command")
+        cmd_subparsers.add_parser("remove",
+                                  help="remove something",
+                                  parents=[parent_parser])
+
+        cmd_subparsers.add_parser("a-very-long-command",
+                                  help="command that does something",
+                                  parents=[parent_parser])
+
+        parser_help = main_parser.format_help()
+        self.assertEqual(parser_help, textwrap.dedent(f'''\
+            usage: {main_parser.prog or ' '} [-h] CMD ...
```

Comment:
I meant that `main_parser.prog` is a constant (if you specify `prog=` in the constructor), so you do not need `.format()` or f-string or whatever. You can have a nice literal string here.